src/main/java/io/appium/java_client/remote/AppiumCommandExecutor.java

```diff
@@ -147,6 +204,18 @@ protected HttpClient getClient() {
         return getPrivateFieldValue(HttpCommandExecutor.class, "client", HttpClient.class);
     }
 
+    /**
+     * Override the http client in the HttpCommandExecutor class with a new http client instance with the given URL.
+     * Use the same http client factory and client config if the constructor got them.
+     *
+     * @param serverUrl to set the URL as the new client's base url.
+     */
+    protected void overrideServerUrl(URL serverUrl) {
+        setPrivateFieldValue(HttpCommandExecutor.class, "client",
```

Comment:
so it is not possible to replace URL for an exiting client without replacing the client instance itself?

Reply:
I haven't dug into netty client itself yet, but maybe we can find a way to replace url only for netty client, but not for client defined by user via factory,
https://github.com/SeleniumHQ/selenium/blob/9656da70e574a77213cba49ff00e11a2d0f1f393/java/src/org/openqa/selenium/remote/http/netty/NettyClient.java
https://github.com/SeleniumHQ/selenium/blob/9656da70e574a77213cba49ff00e11a2d0f1f393/java/src/org/openqa/selenium/remote/http/netty/NettyHttpHandler.java#L56

or maybe replacing ClientConfig instance might work
https://github.com/SeleniumHQ/selenium/blob/9656da70e574a77213cba49ff00e11a2d0f1f393/java/src/org/openqa/selenium/remote/http/RemoteCall.java